plugins/inventory/flightctl.py

```
@@ -591,13 +592,56 @@ def _build_auth_headers(config: Configuration) -> Dict[str, str] | None:
+    while True:
+        try:
+            response = list_func(
+                var_continue=continue_token,
+                label_selector=label_list,
+                field_selector=field_list,
+                limit=limit,
+                _headers=headers,
+                _request_timeout=request_timeout,
+            )
+        except Exception as e:
+            raise FlightctlApiException(f"Error retrieving data from Flight Control API: {e}") from e
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)